lms/djangoapps/student_dashboard/test/test_views.py

```diff
@@ -0,0 +1,111 @@
+        self.request.user = self.user
+        self.course = CourseFactory.create()
+        self.course_enrollments = [
+            CourseEnrollmentFactory.create(user=self.user, course_id=self.course.id, is_active=True,
+                                           course__start=datetime.utcnow() - timedelta(days=2),
+                                           course__end=datetime.utcnow() + timedelta(days=2))]
+        self.course_card = CourseCardFactory(course_id=self.course.id, course_name=self.course.name)
+        self.course_overview = CourseOverviewFactory(display_name='test_overview',
+                                                     **TestStudenDashboard.course_overview_kwargs)
+
+        self.orphan_course = CourseFactory.create()
+        self.orphan_course_overview = CourseOverviewFactory(display_name='orphan_test_overview',
+                                                            id=self.orphan_course.id,
+                                                            **TestStudenDashboard.course_overview_kwargs)
+        self.orphan_course_card = CourseCardFactory(course_id=self.orphan_course.id,
+                                                    course_name=self.orphan_course.name)
+
+        self._init()
+
+    def _init(self):
+        """
+        _init method is for data set initialization.
+        """
+        self.user.extended_profile.interests = TestStudenDashboard.interests
+        self.user.extended_profile.save()
+
+        _settings, _created = CustomSettings.objects.get_or_create(id=self.course.id)
+        _orphan_settings, _orphan_created = CustomSettings.objects.get_or_create(id=self.orphan_course.id)
+
+        _settings.tags = \
+            _orphan_settings.tags = 'Strategy and planning|Leadership and governance|Program design and development'
+
+        _settings.save()
+        _orphan_settings.save()
+
+        initialize_course_settings(self.course, self.course_overview, True)
+        initialize_course_settings(self.orphan_course, self.orphan_course_overview, True)
+
+    @ddt.data(200, 500)
+    def test_get_joined_communities(self, status):
+        """
+        test_get_joined_communities test method is covering
+        lms.djangoapps.student_dashboard.views.get_joined_communities
+        """
+        with mock.patch.object(ForumCategory, 'joined') as joined:
+            joined.return_value = status, []
+            communities = get_joined_communities('some-username')
+            self.assertEqual(communities, [])
+
+    @ddt.data('onboarding', 'other')
+    def test_get_recommended_xmodule_courses(self, _from):
+        """
+             test_get_recommended_xmodule_courses is covering
+             lms.djangoapps.student_dashboard.views.get_recommended_xmodule_courses
+        """
+        recommended_courses = get_recommended_xmodule_courses(self.request, _from)
+        self.assertEqual(len(recommended_courses), 1)
+
+    def test_get_enrolled_past_courses(self):
```

Comment:
Let's please test following cases in separate tests

- When some enrollment course is not in course card list, and when it is.
- When a course has ended when it's not.
- When course card is not in past course and when it is in past courses.

So basically all possible path in the method.

Reply:
Can we please specify the case were testing in the name of this test?